File: Logic Building Problems/Basic Problems/Swap Two Numbers/Swap Two Numbers Without Using Third Variable/method2.java
// Java Code to swap two numbers using bitwise XOR

class method2 {
    public static void main(String[] args) {
        int a = 2, b = 3;
        System.out.println("a = " + a + " b = " + b);   
      
        a = a ^ b;
        b = a ^ b;
        a = a ^ b;
      
        System.out.println("a = " + a + " b = " + b);
    }
}

File: Logic Building Problems/Basic Problems/Swap Two Numbers/Swap Two Numbers Without Using Third Variable/method2.js
// JavaScript Code to swap two numbers using bitwise XOR

let a = 2, b = 3;
console.log("a = " + a + " b = " + b);   

a = a ^ b;
b = a ^ b;
a = a ^ b;

console.log("a = " + a + " b = " + b);